includes/class-admin.php

```diff
@@ -9,17 +9,176 @@
  * Adds WP-Admin functionality
  *
  * Class Admin
+ *
  * @package WP_SUB
  */
 class Admin {
 
+	/**
+	 * @return void
+	 */
 	public function register_hooks() {
 		// Network admin user table integration
 		add_filter( 'manage_users-network_columns', array( $this, 'add_network_column' ) );
 		add_action( 'manage_users_custom_column', array( $this, 'render_network_column' ), 10, 3 );
 
 		// Site admin user table integration
-		add_action( 'load-users.php', [ $this, 'on_user_page_init' ] );
+		add_action( 'load-users.php', array( $this, 'on_user_page_init' ) );
+
+		// Site admin user add table manage site
+		add_action( 'edit_user_profile', array( $this, 'manage_user_section' ), 1 );
+		add_action( 'show_user_profile', array( $this, 'manage_user_section' ), 1 );
+		add_action( 'personal_options_update', array( $this, 'manage_user_section_update' ) );
+		add_action( 'edit_user_profile_update', array( $this, 'manage_user_section_update' ) );
+
+		// add js file
+		add_action( 'admin_enqueue_scripts', array( $this, 'table_enqueue_admin_script' ) );
+	}
+
+	public function table_enqueue_admin_script( $hook ) {
+		wp_enqueue_script( 'script_table', plugin_dir_url( __FILE__ ) . 'js/table.js', array(), '1.0' );
+	}
+
+	public function manage_user_section( $user ) {
+		if ( user_can( $user->ID, 'manage_network_users' ) ) {
+			$this->manage_site_options( $user );
+			$this->manage_network_options( $user );
+		}
+	}
+
+	/**
+	 * @return void
+	 */
+	protected function manage_site_options( $user ) {
+		$all_sites         = get_sites();
+		$current_user_site = wp_sub_get_user_sites( $user->ID );
+		?>
+		<h2><?php esc_html_e( 'Manage Sites', 'wp-separate-user-base' ); ?></h2>
+		<?php
+		foreach ( $all_sites as $site ) {
+			$site->site_name = $site->blogname;
+			$site->id        = $site->blog_id;
+		}
+		$this->create_table( 'site_id', $all_sites, $current_user_site );
+	}
+
+	/**
+	 * @return void
+	 */
+	protected function manage_network_options( $user ) {
+		$all_networks         = get_networks();
+		$current_user_network = get_user_meta( $user->ID, WP_Separate_User_Base::NETWORK_META_KEY, false );
```

Comment:
Can you please add a similar function as `wp_sub_get_user_sites' in the functions.php file to keep things consistent. So `wp_sub_get_user_networks`